### PR TITLE
minor flags refactoring

### PR DESCRIPTION
**Pull request by yonromai**

### CODE REVIEW

**Comment by yonromai** [comment]
@ravwojdyla PTAL

File: spotify_tensorflow/flags.py [hunk 1]
```diff
@@ -30,20 +31,35 @@ class Flags(object):
...
     def register_dataset_flags():
         logging.info("Registering Dataset flags")
+
+        flags.DEFINE_string("train_subdir", "train",
+                            "Location of training TFRecords, with the training set dir.")
+
+        flags.DEFINE_string("eval_subdir", "eval",
+                            "Location of eval TFRecords, with the training set dir.")
+
+        # Dataset API parameters
```
Comment (ravwojdyla) [inline spotify_tensorflow/flags.py]: I wonder whether we should have a namespace here? Like `datasets_ batch_size` and `datasets_ prefetch_buffer_size`? What do you think?

File: spotify_tensorflow/flags.py [hunk 2]
```diff
@@ -30,20 +31,35 @@ class Flags(object):
...
                              "Creates a `Dataset` with at most `count` batches from this dataset.")
 
-        flags.DEFINE_string("train_subdir", "train",
-                            "Location of training TFRecords, with the training set dir.")
+        flags.DEFINE_integer("parsing_threads", mp.cpu_count(),
+                             "Number of threads used for parsing files.")
 
-        flags.DEFINE_string("eval_subdir", "eval",
-                            "Location of eval TFRecords, with the training set dir.")
+        flags.DEFINE_integer("interleaving_threads", 2,
```
Comment (ravwojdyla) [inline spotify_tensorflow/flags.py]: I wonder if this could be a function of `cpu_count` like `floor(cpu_count/2)`?

Comment (yonromai) [reply spotify_tensorflow/flags.py]: I haven't played with other machines but I found this parameter to be extremely sensitive, like going from 2 to 3 lower IO throughput on my machine by 200%. 
I think we should be conservative with this one until further consideration. (I was actually quite tempted to turn it off by default)

File: spotify_tensorflow/flags.py (hunk 1 above)
Comment (yonromai) [reply spotify_tensorflow/flags.py]: I don't think it's super necessary, for example the meaning of the `batch_size` parameter totally align with what ML ppl think this parameter means.

**Pull request merged**